protocol/activity_center_persistence.go

```diff
@@ -446,23 +468,34 @@ func (db sqlitePersistence) GetNotReadActivityCenterNotificationIds() ([][]byte,
 }
 
 func (db sqlitePersistence) GetToProcessActivityCenterNotificationIds() ([][]byte, error) {
-	return db.runActivityCenterIDQuery("SELECT a.id FROM activity_center_notifications a WHERE NOT a.dismissed AND NOT a.accepted")
+	return db.runActivityCenterIDQuery(`
+		SELECT a.id
+		FROM activity_center_notifications a
+		WHERE NOT a.deleted AND NOT a.dismissed AND NOT a.accepted
+		`)
 }
 
 func (db sqlitePersistence) HasPendingNotificationsForChat(chatID string) (bool, error) {
-	rows, err := db.db.Query("SELECT 1 FROM activity_center_notifications a WHERE a.chat_id = ? AND NOT a.dismissed AND NOT a.accepted", chatID)
+	rows, err := db.db.Query(`
+		SELECT 1 FROM activity_center_notifications a
```

Comment:
Maybe we want to check indexes on this query? (if they need to be modified/added etc)

Reply:
Thanks for bringing this up. I did a rough investigation on how/if indexes are used for some queries related to the `activity_center_notifications` table.

SQLite is doing a full table scan when finding pending notifications. Once I add the following index, then it reports what we want:

```sql
CREATE INDEX activity_center_chat_id_deleted_dismissed_accepted_idx
  ON activity_center_notifications(chat_id, deleted, dismissed, accepted);

EXPLAIN QUERY PLAN
SELECT * FROM activity_center_notifications
WHERE chat_id = 'test-chat'
  AND NOT deleted
  AND NOT dismissed
  AND NOT accepted;

-- Reports:
-- SEARCH activity_center_notifications USING INDEX activity_center_chat_id_deleted_dismissed_accepted_idx (chat_id=?)
```

Currently, there are only two indexes, and they don't cover the query above, so I'm going to add the index to the migration.

- `activity_center_dimissed_accepted` (yes, there's a typo)
- `activity_center_read`
- The automatic index for the primary key

Reply:
@cammellos, what do you think about the solution? Can I resolve this thread and the other ones about indexes or do you have other concerns related to that?